website/src/content/docs/docs/howtos/ARM/versioning.md

```diff
@@ -0,0 +1,315 @@
+  @removed(Versions.v3)
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<Employee>;
+}
+```
+
+### @madeRequired and @madeOptional
+
+Use these decorators to change whether a property is required or optional in a specific version.
+
+- Use `@madeOptional(Versions.version)` to make a property optional starting in that version.
+- Use `@madeRequired(Versions.version)` to make a property required starting in that version.
+
+**Example: Changing a property from required to optional, then back to required**
+
+For example:
+
+```tsp
+// v1: movingStatus is required
+model MoveResponse {
+  movingStatus: string;
+}
+
+// v2: movingStatus becomes optional
+model MoveResponse {
+  @madeOptional(Versions.v2)
+  movingStatus?: string;
+}
+```
+
+### @renamedFrom
+
+Use `@renamedFrom` to rename models, properties, operations, enums, etc., in a specific version.
+
+- The `version` argument is the version where the name changed.
+- The `oldName` argument is the previous name.
+
+**Example: Renaming a property and a model across versions**
+
+For example:
+
+```tsp
+// v1: Model and property have original names
+model WorkerProperties {
+  state?: string;
+}
+
+// v2: Rename property 'state' to 'city'
+model WorkerProperties {
+  @renamedFrom(Versions.v2, "state")
+  city?: string;
+}
+
+// v3: Rename model 'WorkerProperties' to 'EmployeeProperties'
+@renamedFrom(Versions.v3, "WorkerProperties")
+model EmployeeProperties {
+  @renamedFrom(Versions.v2, "state")
+  city?: string;
+}
+```
+
+### @returnTypeChangedFrom
+
+Use this decorator to change the return type of an operation in a specific version.
+
+```tsp
+@armResourceOperations
+interface Employees {
+  @removed(Versions.v2)
+  @returnTypeChangedFrom(Versions.v2, Worker)
+  get is ArmResourceRead<Employee>;
+}
+
+model Employee {
+  name: string;
+}
+
+model Worker {
+  name: string;
+}
+```
+
+## Complex Scenarios
+
+---
+
+### Adding Decoration to an Existing Type
+
+This scenario demonstrates how to change a decorator’s usage across versions.
+
+Suppose you have a model property with a decorator:
+
+```tsp
+model Employee {
+  @visibility(Lifecycle.Read)
+  experience: string;
+}
+```
+
+In the next version (v2), the visibility changes so that the property can be read or created. To achieve this, you can use a combination of `@removed`, `@added`, and `@renamedFrom` decorators:
+
+```tsp
+model Employee {
+  @removed(Versions.v2)
+  @visibility(Lifecycle.Read)
+  @renamedFrom(Versions.v2, "experience")
+  oldExperience: string;
+
+  @added(Versions.v2)
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  experience: string;
+}
+```
+
+### Adding a Parameter to an Operation and Making Another Parameter Optional
+
+This scenario shows how to add a parameter and make another optional in an operation.
+
+Suppose you start with the following operation in v1:
+
+```tsp
+@armResourceOperations
+interface Employees {
+  get(name: string, identifier: int32): Employee | ErrorResponse;
```

Comment:
These examples should all use Arm Templates.  The 'Parameters' parameter allows you to add query and header parameters to an operation, which is what this should mainly cover.